common/src/main/java/com/wynntils/core/features/overlays/OverlayManager.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.features.overlays;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.features.overlays.annotations.OverlayInfo;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.mc.utils.McUtils;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class OverlayManager {
+    private static final Map<Overlay, OverlayInfo> overlayInfoMap = new HashMap<>();
+
+    private static final Set<Overlay> enabledOverlays = new HashSet<>();
+
+    public static void registerOverlay(Overlay overlay, OverlayInfo overlayInfo) {
+        overlayInfoMap.put(overlay, overlayInfo);
+
+        if (overlayInfo.enabled()) {
+            enabledOverlays.add(overlay);
+        }
+    }
+
+    public static void disableAllOverlaysForFeature(List<Overlay> overlays) {
```

Comment:
I think the method should be renamed to just `disableOverlays`. (And similar for enable)

An alternative would be to actually send in a Feature, but then the logic of extracting the set of overlays in the feature need to be either moved here, or exposed in the Feature class, and neither seems really good.

And it is more general to just take a list of overlays and disable them, as you've done.

Reply:
Yeah, I think it was my purpose to rename it but after I changed up the logic, I forgot.😅